department-of-veterans-affairs/caseflow
# CASEFLOW-93 #done Automatically place CAVC Remands without mandate on hold

Resolves [CASEFLOW-93](https://vajira.max.gov/browse/CASEFLOW-93)

As a CAVC Litigation Support user, I need a new task type to be created for MDR CAVC Remands, so that the task can be placed on hold until I need to continue processing the case.

### Description
Automatically place CAVC Remands without mandate on hold

### Acceptance Criteria
- [x] This feature should be accessible to the following user groups: CAVC Litigation Support
- [x]  Include screenshot(s) in the Github issue if there are front-end changes
- [x]  Update documentation: [CAVC Remand](https://github.com/department-of-veterans-affairs/caseflow/wiki/CAVC-Remands)
- [x]  Verify new task type MDR Task is created
- [x]  Verify the task is assigned to the CAVC Litigation Support team
- [x]  Verify the task is child of CAVC task on new appeal stream
- [x]  Verify the task is automatically placed on hold for 90 days
- [x]  Verify the following action is available on the task
   - [x] End hold early

### Testing Plan
1. Open a Rails console and create a ready for CAVC appeal, note the `stream_docket_number` so you can search for the appeal in the UI or the `uuid`.
```ruby
original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first)
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes")
```
2. Login as a CAVC lit support user
3. Search for the dispatched appeal you just created and select it to go to Case Details: `http://localhost:3000/queue/appeals/<UUID>`
4. Click the "+ Add CAVC Remand" button in the "Post-Despatch actions" section
5. Fill out the "Add a Court Remand" form and submit. (Note that an MDR remand does not require entry of a judgement or mandate date however they may be entered if they do exist. We should probably hide those dates.)
6. Copy the UUID of the newly created cavc appeal and check out its task tree in the console
```ruby
cavc_appeal=Appeal.find_by_uuid("be9b6b0f-e2cc-4617-b2e7-f8a24f121da2")
cavc_appeal.treee
                                          ┌─────────────────────────────────────────────────────────────────────────────┐
Appeal 1039 (E 210120-1038 Court Remand)  │ ID   │ STATUS   │ ASGN_BY │ ASGN_TO               │ UPDATED_AT              │
└── RootTask                              │ 3428 │ on_hold  │         │ Bva                   │ 2021-01-21 19:57:47 UTC │
    └── DistributionTask                  │ 3429 │ on_hold  │         │ Bva                   │ 2021-01-21 19:57:47 UTC │
        └── CavcTask                      │ 3430 │ on_hold  │         │ Bva                   │ 2021-01-21 19:57:47 UTC │
            └── MdrTask                   │ 3431 │ on_hold  │         │ CavcLitigationSupport │ 2021-01-21 19:57:47 UTC │
                └── TimedHoldTask         │ 3432 │ assigned │         │ CavcLitigationSupport │ 2021-01-21 19:57:47 UTC │
                                          └─────────────────────────────────────────────────────────────────────────────┘
```
or in the browser: `http://localhost:3000/task_tree/appeals/be9b6b0f-e2cc-4617-b2e7-f8a24f121da2`
7. Test the End hold early action and check the task tree
8. Play around, stress test, and find bugs early

### User Facing Changes
 - [x] Screenshots of UI changes
 
 BEFORE|AFTER
 ---|---
(No actions) | ![image](https://user-images.githubusercontent.com/55255674/105405950-5ec2c200-5bf1-11eb-88f8-dfca705f7b23.png)


### Code Documentation Updates
